x-pack/plugins/fleet/server/services/fleet_server_migration.ts

```diff
@@ -76,24 +148,61 @@ async function migrateEnrollmentApiKeys() {
       hasMore = false;
     }
     for (const item of res.items) {
-      const key = await getEnrollmentAPIKey(soClient, item.id);
-
-      const body: FleetServerEnrollmentAPIKey = {
-        api_key: key.api_key,
-        api_key_id: key.api_key_id,
-        active: key.active,
-        created_at: key.created_at,
-        name: key.name,
-        policy_id: key.policy_id,
-      };
-      await esClient.create({
-        index: ENROLLMENT_API_KEYS_INDEX,
-        body,
-        id: key.id,
-        refresh: 'wait_for',
-      });
+      try {
+        const key = await getEnrollmentAPIKey(soClient, item.id);
+
+        const body: FleetServerEnrollmentAPIKey = {
+          api_key: key.api_key,
+          api_key_id: key.api_key_id,
+          active: key.active,
+          created_at: key.created_at,
+          name: key.name,
+          policy_id: key.policy_id,
+        };
+        await esClient.create({
+          index: ENROLLMENT_API_KEYS_INDEX,
+          body,
+          id: key.id,
+          refresh: 'wait_for',
+        });
 
-      await soClient.delete(ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE, key.id);
+        await soClient.delete(ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE, key.id);
+      } catch (error) {
+        // swallow 404 error has multiple Kibana can run the migration at the same time
+        if (!is404Error(error)) {
+          throw error;
+        }
+      }
     }
   }
 }
+
+async function migrateAgentPolicies() {
+  const esClient = appContextService.getInternalUserESClient();
+  const soClient = getInternalUserSOClient();
+  const { items: agentPolicies } = await agentPolicyService.list(soClient, {
+    perPage: SO_SEARCH_LIMIT,
+  });
+
+  await Promise.all(
+    agentPolicies.map(async (agentPolicy) => {
+      const res = await esClient.search({
```

Comment:
this might be able to be made more performant to by doing a single search call, or maybe even msearch. I'm looking closely at the performance aspect of these migrations because of the past issues we've had with Fleet setup timing out